hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -56,14 +84,203 @@ public void preHandle(@NonNull final PreHandleContext context) {
+        if (optionalPayer.isEmpty()) {
+            throw new HandleException(INVALID_PAYER_ACCOUNT_ID);
+        }
+        final var payer = optionalPayer.get();
+        final long newPayerBalance = payer.tinybarBalance() - op.initialBalance();
+        validatePayer(payer, newPayerBalance);
+
+        // Change payer's balance to reflect the deduction of the initial balance for the new
+        // account
+        final var modifiedPayer =
+                payer.copyBuilder().tinybarBalance(newPayerBalance).build();
+        accountStore.put(modifiedPayer);
+
+        // Build the new account to be persisted based on the transaction body
+        final var accountCreated = buildAccount(op, handleContext);
+        accountStore.put(accountCreated);
+
+        // set newly created account number in the record builder
+        final var createdAccountNum = accountCreated.accountNumber();
+        recordBuilder.setCreatedAccount(createdAccountNum);
+
+        // put if any new alias is associated with the account into account store
+        if (op.alias() != Bytes.EMPTY) {
+            accountStore.putAlias(op.alias().toString(), createdAccountNum);
+        }
+    }
+
+    @Override
+    public CryptoCreateRecordBuilder newRecordBuilder() {
+        return new CreateAccountRecordBuilder();
+    }
+
+    /* ----------- Helper Methods ----------- */
+
+    /**
+     * Validate the basic fields in the transaction body that does not involve checking with dynamic
+     * properties or state. This check is done as part of the pre-handle workflow.
+     * @param op the transaction body
+     * @return OK if the transaction body is valid, otherwise return the appropriate error code
+     */
+    private ResponseCodeEnum pureChecks(@NonNull final CryptoCreateTransactionBody op) {
+        if (op.initialBalance() < 0L) {
+            return INVALID_INITIAL_BALANCE;
+        }
+        if (!op.hasAutoRenewPeriod()) {
+            return INVALID_RENEWAL_PERIOD;
+        }
+        if (op.sendRecordThreshold() < 0L) {
+            return INVALID_SEND_RECORD_THRESHOLD; // should this return SEND_RECORD_THRESHOLD_FIELD_IS_DEPRECATED
+        }
+        if (op.receiveRecordThreshold() < 0L) {
+            return INVALID_RECEIVE_RECORD_THRESHOLD; // should this return RECEIVE_RECORD_THRESHOLD_FIELD_IS_DEPRECATED
+        }
+        if (op.hasProxyAccountID() && !op.proxyAccountID().equals(AccountID.DEFAULT)) {
+            return PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED;
+        }
+        return OK;
+    }
+
+    /**
+     * Validate the fields in the transaction body that involves checking with dynamic
+     * properties or state. This check is done as part of the handle workflow.
+     * @return OK if the transaction body is valid, otherwise return the appropriate error code
+     */
+    private ResponseCodeEnum validateSemantics() {
+        // TODO : Need to add validations that involve dynamic properties or state
```

Comment:
Not sure if you already planned, but I think it makes sense to do this as a separate PR, given the complexity involved.

Reply:
yeah planned for different PR 😁